trie/trie.go

```diff
@@ -287,34 +380,46 @@ func concat(s1 []byte, s2 ...byte) []byte {
 	return r
 }
 
-func (t *Trie) resolve(n node) node {
+func (t *Trie) resolve(n node, prefix, suffix []byte) (node, error) {
 	if n, ok := n.(hashNode); ok {
-		return t.resolveHash(n)
+		return t.resolveHash(n, prefix, suffix)
+	}
+	return n, nil
+}
+
+type MissingNodeError struct {
+	RootHash, NodeHash   common.Hash
+	KeyPrefix, KeySuffix []byte
+}
```

Comment:
Any particular reason this is in the middle of the file? :P 

Please either move this to the top of the file, or better yet I would create an errors.go file to contain this. Usually that is the convention throughout our code base.